uaclient/actions.py

```diff
@@ -161,6 +162,34 @@ def attach_with_token(
 
     machine_token_file.write(new_machine_token)
 
+    current_series = system.get_release_info().series
+    if current_series:
+        support_entitlement = (
+            machine_token_file.entitlements()
+            .get("support", {})
+            .get("entitlement")
+        )
+        if support_entitlement:
+            allowed_series = support_entitlement.get("affordances", {}).get(
+                "onlySeries"
+            )
+            if allowed_series and allowed_series != "none":
+                try:
+                    allowed_release = CODENAME_TO_RELEASE.get(allowed_series)
+                except KeyError:
+                    allowed_release = ""
```

Comment:
`dict.get()` will never raise a KeyError. It will return `None` if the key isn't present by default, but you can change the default value to `""`
```suggestion
                allowed_release = CODENAME_TO_RELEASE.get(allowed_series, "")
```